README.md

```diff
@@ -65,6 +64,38 @@ int main()
 
     // Send a metric explicitly
     client.send("tutu", 4, "c", 2.0f);
+    exit(0);
+}
+```
+
+### Advanced Testing
+
+A simple mock StatsD server can be found at `tests/StatsdServer.hpp`. This can be used to do simple validation of your application's metrics, typically in the form of unit tests. In fact this is the primary means by which this library is tested. The mock server itself is not distributed with the library so to use it you'd need to vendor this project into your project. Once you have though, you can test your application's use of the client like so:
```

Comment:
here ive implemented the first option you suggested which was to just document the existance of the mock server.

a second option to avoid having to distribute the mock server as another header would be to put it into the `UDPSender.hpp` header and have it share some of its common bits with it the sender. this would both reduce duplicated code and make the mock receiver accesable to downstream projects.

the third option would be to hide it completely and remove this entry from the readme and just have it defined and used in the test applications code.

@vthiery let me know what you like best of the 3 options

Reply:
I prefer option one for the time being. Thanks for the doc!